regression/cbmc/nondet-pointer/nondet-pointer2.c

```diff
@@ -0,0 +1,14 @@
+int *nondet_pointer();
+
+int main()
+{
+  int x = 123, y = 456;
+  int *px = &x;
+  int *py = &y;
+  int *q = nondet_pointer();
+
+  if(q == px || q == py)
```

Comment:
Why is it sufficient to test if q == px or q == py? Should we not also have to test same_object? 
Why do I say that? Daniel has pointed out to me that we can construct two pointers that are equal but have different provenance and accessing one may cause an error.
Suppose y comes after x in memory. Suppose we choose q to be derived from px (&x) but with an offset large enough to point it at y. Now q == py ... but is it legal to dereference it and get 456, or should I expect failure?

Reply:
Nondeterministic pointers are outside of the semantics of C; we can make that go either way.